commands/api-gateway/gateway/customDomains/add.go

```diff
@@ -0,0 +1,93 @@
+package customDomains
+
+import (
+	"context"
+	"fmt"
+	"github.com/ionos-cloud/ionosctl/v6/commands/api-gateway/completer"
+	"github.com/ionos-cloud/ionosctl/v6/internal/client"
+	"github.com/ionos-cloud/ionosctl/v6/internal/constants"
+	"github.com/ionos-cloud/ionosctl/v6/internal/core"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/json2table/resource2table"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/jsontabwriter"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/tabheaders"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/pointer"
+	"github.com/ionos-cloud/sdk-go-bundle/products/apigateway/v2"
+	"github.com/spf13/viper"
+)
+
+func AddCmd() *core.Command {
+	cmd := core.NewCommand(context.Background(), nil, core.CommandBuilder{
+		Namespace: "apigateway",
+		Resource:  "customdomains",
+		Verb:      "add",
+		Example:   "ionosctl apigateway gateway customdomains add --gateway-id ID --name NAME --certificate-id CERTIFICATEID",
+		PreCmdRun: func(c *core.PreCommandConfig) error {
+			if err := core.CheckRequiredFlags(c.Command, c.NS, constants.FlagGatewayID, constants.FlagName, constants.FlagCertificateId); err != nil {
+				return err
+			}
+
+			return nil
+		},
+		CmdRun: func(c *core.CommandConfig) error {
+			apigatewayId := viper.GetString(core.GetFlagName(c.NS, constants.FlagGatewayID))
+			usedGateway, _, err := client.Must().Apigateway.APIGatewaysApi.ApigatewaysFindById(context.Background(), apigatewayId).Execute()
+			input := usedGateway.Properties
+			elem := len(input.CustomDomains)
+			if input.CustomDomains == nil {
+				input.CustomDomains = make([]apigateway.GatewayCustomDomains, 1)
+			}
+			input.CustomDomains = append(input.CustomDomains, apigateway.GatewayCustomDomains{})
+			elem = len(input.CustomDomains) - 1
```

Comment:
(just a nitpick and can be ignored) 

we don't do anything with this value except re-assign it later. this line can be safely removed

```suggestion
			if input.CustomDomains == nil {
				input.CustomDomains = make([]apigateway.GatewayCustomDomains, 1)
			}
			input.CustomDomains = append(input.CustomDomains, apigateway.GatewayCustomDomains{})
			elem := len(input.CustomDomains) - 1
```